### PR TITLE
Add logarithms with different bases

### LINKED ISSUE #1209: Add Nx.log10 and Nx.log2 functions

### PR DESCRIPTION
**Pull request by msluszniak**
Closes #1209

### CODE REVIEW

File: nx/lib/nx.ex [hunk 1]
```diff
@@ -15412,6 +15412,77 @@ defmodule Nx do
...
+      >
+
+      iex> Nx.log_p(Nx.tensor([3, 9, 27, 81]), 3)
+      #Nx.Tensor<
+        f32[4]
+        [1.0, 2.0, 3.0, 4.0]
+      >
+  """
+  @doc type: :element
+  def log_p(tensor, base) do
```
Comment (polvalente) [inline nx/lib/nx.ex:15477]: is log_p the Jax name for this?

Comment (polvalente) [reply nx/lib/nx.ex:15477]: Couldn't we have log/2 that accepts the base optionally?

Comment (msluszniak) [reply nx/lib/nx.ex:15477]: > Couldn't we have log/2 that accepts the base optionally?

It sounds good to me.

Comment (josevalim) [reply nx/lib/nx.ex:15477]: No need, you can define `log(..., ...)` that takes two args, works completely separate and it works similar to the versions you defined here.

File: nx/lib/nx.ex [hunk 2]
```diff
@@ -15412,6 +15412,77 @@ defmodule Nx do
...
+
+      iex> Nx.log(Nx.tensor([3, 9, 27, 81]), 3)
+      #Nx.Tensor<
+        f32[4]
+        [1.0, 2.0, 3.0, 4.0]
+      >
+  """
+  @doc type: :element
+  def log(tensor, base) do
+    if base <= 0 or base == 1 do
```
Comment (josevalim) [inline nx/lib/nx.ex:15478]:
```suggestion
    if is_integer(base) and (base <= 0 or base == 1) do
```

File: nx/lib/nx.ex [hunk 3]
```diff
@@ -15412,6 +15412,77 @@ defmodule Nx do
...
+        [0.0, 1.0, 2.0, 3.0]
+      >
+  """
+  @doc type: :element
+  def log10(tensor) do
+    divide(log(tensor), log(10))
+  end
+
+  @doc """
+  Calculates the element-wise logarithm of a tensor with given base `p`.
```
Comment (josevalim) [inline nx/lib/nx.ex:15460]:
```suggestion
  Calculates the element-wise logarithm of a tensor with given `base`.
```

File: nx/lib/nx.ex (hunk 2 above)
Comment (msluszniak) [reply nx/lib/nx.ex:15478]: base is not and integer in general like in `log`

Comment (polvalente) [reply nx/lib/nx.ex:15478]: Base can be any non negative number, right?

Comment (polvalente) [reply nx/lib/nx.ex:15478]: Nvm, log(1) == 0 haha

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**